### PR TITLE
Train entrypoint refactoring + misc minor refactoring 

### PR DESCRIPTION
**Pull request by AlekseySh**

### CODE REVIEW

File: oml/lightning/entrypoints/parser.py [hunk 1]
```diff
@@ -49,4 +60,104 @@ def check_is_config_for_ddp(cfg: TCfg) -> bool:
...
+    sampler_runtime_args = {"labels": dataset.get_labels(), "label2category": dataset.get_label2category()}
+    sampler = get_sampler_by_cfg(cfg["sampler"], **sampler_runtime_args) if cfg["sampler"] is not None else None
+
+    return sampler
+
+
+def parse_ckpt_callback_from_config(cfg: TCfg) -> ModelCheckpoint:
+    return ModelCheckpoint(
+        dirpath=Path.cwd() / "checkpoints",
+        monitor=cfg.get("metric_for_checkpointing", f"{OVERALL_CATEGORIES_KEY}/cmc/1"),
```
Comment (DaloroAT) [inline oml/lightning/entrypoints/parser.py:147]: is it good default? For benchmarking it's ok, but in practice, no one should be interest this metric))

File: oml/lightning/entrypoints/parser.py [hunk 2]
```diff
@@ -49,4 +60,104 @@ def check_is_config_for_ddp(cfg: TCfg) -> bool:
...
+
+    return logger
+
+
+def upload_files_to_neptune_cloud(logger: NeptuneLogger, cfg: TCfg) -> None:
+    assert isinstance(logger, NeptuneLogger)
+
+    # save transforms as files
+    for key, val in cfg.items():
+        if "transforms" in key:
```
Comment (DaloroAT) [inline oml/lightning/entrypoints/parser.py:97]: also we have `transforms_val`, `transforms_train`

File: oml/lightning/entrypoints/parser.py [hunk 3]
```diff
@@ -49,4 +60,104 @@ def check_is_config_for_ddp(cfg: TCfg) -> bool:
...
+            "scheduler_frequency": cfg["scheduling"]["scheduler_frequency"],
+            "scheduler_monitor_metric": cfg["scheduling"].get("monitor_metric", None),
+        }
+    else:
+        scheduler_kwargs = {"scheduler": None}
+
+    return scheduler_kwargs
+
+
+def parse_sampler_from_config(cfg: TCfg, dataset: DatasetWithLabels) -> IBatchSampler:
```
Comment (DaloroAT) [inline oml/lightning/entrypoints/parser.py:131]: wrong typing for output

File: oml/lightning/entrypoints/parser.py [hunk 4]
```diff
@@ -49,4 +60,104 @@ def check_is_config_for_ddp(cfg: TCfg) -> bool:
...
+    If Neptune Logger is used, we also upload to its cloud some files
+    which are good to have for reproducibility.
+
+    """
+    pprint(cfg)
+
+    if ("NEPTUNE_API_TOKEN" in os.environ.keys()) and (cfg["neptune_project"] is not None):
+        cwd = Path.cwd()
+        logger = NeptuneLogger(
+            api_key=os.environ["NEPTUNE_API_TOKEN"],
```
Comment (DaloroAT) [inline oml/lightning/entrypoints/parser.py:76]: previously we tried to read config from hydra config

File: oml/lightning/entrypoints/train.py [hunk 5]
```diff
@@ -128,33 +51,65 @@ def pl_train(cfg: TCfg) -> None:
...
+    loader_train, loaders_val = get_retrieval_loaders(cfg)
+    extractor = get_extractor_by_cfg(cfg["model"])
+    criterion = get_criterion_by_cfg(cfg["criterion"], **{"label2category": loader_train.dataset.get_label2category()})
+    optimizable_parameters = [
+        {"lr": cfg["optimizer"]["args"]["lr"], "params": extractor.parameters()},
+        {"lr": cfg["optimizer"]["args"]["lr"], "params": criterion.parameters()},
+    ]
+    optimizer = get_optimizer_by_cfg(cfg["optimizer"], **{"params": optimizable_parameters})  # type: ignore
+
+    module_kwargs = parse_scheduler_from_config(cfg, optimizer=optimizer)
```
Comment (DaloroAT) [inline oml/lightning/entrypoints/train.py:88]: `module_kwargs = parse_scheduler_from_config` looks strange in terms of naming. But probably we don't want to create an empty dict and fill it

**Review by DaloroAT — APPROVED** [review]

File: oml/lightning/entrypoints/parser.py (hunk 3 above)
Comment (AlekseySh) [reply oml/lightning/entrypoints/parser.py:131]: okay

**Pull request merged**